plugins/adr/src/components/AdrReader/AdrReader.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useMemo, useState } from 'react';
+import {
+  InfoCard,
+  MarkdownContent,
+  Progress,
+  WarningPanel,
+} from '@backstage/core-components';
+import { getAdrLocationUrl } from '@backstage/plugin-adr-common';
+import { useEntity } from '@backstage/plugin-catalog-react';
+
+import { useOctokitRequest } from '../../hooks';
+import { adrDecoratorFactories } from './decorators';
+import { AdrContentDecorator } from './types';
+
+/**
+ * Component to fetch and render an ADR.
+ * @public
+ */
+export const AdrReader = ({
+  adr,
+  decorators,
+}: {
+  adr: string;
+  decorators?: AdrContentDecorator[];
+}) => {
+  const { entity } = useEntity();
+  const [adrContent, setAdrContent] = useState('');
+  const adrLocationUrl = getAdrLocationUrl(entity);
+
+  const { value, loading, error } = useOctokitRequest(
+    `${adrLocationUrl}/${adr}`,
+  );
+
+  const adrDecorators = useMemo(
+    () =>
+      decorators ?? [
+        adrDecoratorFactories.createRewriteRelativeLinksDecorator(),
+        adrDecoratorFactories.createRewriteRelativeEmbedsDecorator(),
+      ],
+    [decorators],
+  );
+
+  useEffect(() => {
+    if (!value?.data) {
+      return;
+    }
+
+    setAdrContent(
+      adrDecorators.reduce(
+        (content, decorator) => decorator({ baseUrl: adrLocationUrl, content }),
+        value.data,
+      ),
+    );
+  }, [adrDecorators, adrLocationUrl, value]);
```

Comment:
Prolly better to compute this with `useMemo` instead of `useState` + `useEffect` here I think